### PR TITLE
fix: MarkdownAnchor should still render children

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
### PR Checklist

- [ ] Related to Abstract designs:
- [ ] Related to JIRA ticket: ABC-123
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### Description

This came from a cleanup I'd done post-checking in #883.

### CODE REVIEW

File: packages/gamut/src/Markdown/libs/overrides/MarkdownAnchor/__tests__/MarkdownAnchor-test.tsx [hunk 1]
```diff
@@ -38,4 +38,12 @@ describe('MarkdownAnchor', () => {
     const anchor = mount(<MarkdownAnchor href="www.codecademy.com" />);
     expect(anchor.find(`a[href='www.codecademy.com']`).length).toEqual(1);
   });
+
+  it('renders its children', () => {
+    const text = 'natalie rulez';
+
+    const anchor = mount(<MarkdownAnchor href="/">{text}</MarkdownAnchor>);
```
Comment (jakemhiller) [inline packages/gamut/src/Markdown/libs/overrides/MarkdownAnchor/__tests__/MarkdownAnchor-test.tsx:45]: is there a test for this in the root markdown tests? the integration-style test might be more accurate

Comment (jakemhiller) [reply packages/gamut/src/Markdown/libs/overrides/MarkdownAnchor/__tests__/MarkdownAnchor-test.tsx:45]: looks like the `Markdown anchor links` test should also check for the text content of the tag

Comment (jakemhiller) [reply packages/gamut/src/Markdown/libs/overrides/MarkdownAnchor/__tests__/MarkdownAnchor-test.tsx:45]: https://github.com/Codecademy/client-modules/blob/main/packages/gamut/src/Markdown/__tests__/Markdown-test.tsx/#L179-L186

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.8.6-alpha.86fdd6.0<br/>@codecademy/gamut@12.2.5-alpha.86fdd6.0<br/></pre>

File: packages/gamut/src/Markdown/__tests__/Markdown-test.tsx [hunk 2]
```diff
@@ -269,4 +269,14 @@ var test = true;
       expect(markdown.text().trim()).toEqual(expectedText);
     });
   });
+
+  describe('Renders links', () => {
```
Comment (jakemhiller) [inline packages/gamut/src/Markdown/__tests__/Markdown-test.tsx:273]: prob worth keeping this together w/ the existing anchor tests: https://github.com/Codecademy/client-modules/blob/main/packages/gamut/src/Markdown/__tests__/Markdown-test.tsx#L179

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f2c721d933f0f00bfdb44af--gamut-preview.netlify.app>https://5f2c721d933f0f00bfdb44af--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f2c721d933f0f00bfdb44af>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.8.6-alpha.8fa6e9.0<br/>@codecademy/gamut@12.2.5-alpha.8fa6e9.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f2c73582c6948014fdab7a2--gamut-preview.netlify.app>https://5f2c73582c6948014fdab7a2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f2c73582c6948014fdab7a2>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.8.6-alpha.58a979.0<br/>@codecademy/gamut@12.2.5-alpha.58a979.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f2c73fbb2779d0193c5e5a2--gamut-preview.netlify.app>https://5f2c73fbb2779d0193c5e5a2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f2c73fbb2779d0193c5e5a2>Deploy Logs</a></p>

**Pull request merged**